docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,303 @@
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Client
+  participant S_Auth as Authentification server
+  participant S_Data as Metadata server
+
+  Client  ->> S_Auth: Retrieve the device of the user
+  S_Auth -->> Client: Device of the user
+
+  Client  ->> S_Data: Exchange with the server
+```
+
+### Account creation
+
+Creating the account for a user will require some information for the system to work:
+
+- An identifier for the user (email)
+
+  The service needs to verify that the user has access to the email (using a code or sending a link to continue the process).
+
+- A secret to identify the user (password, Fido2, ...)
+
+  For the password, it's not used as is, but it's derived to create a new secret using the function `PBKDF_A` (this operation is done on the client side to not leak the master password).
+  The new secret will be provided as is to the server (like a password).
+
+  In the FIDO2 case the included challenge is used as is since the private key is not shared with the server.
+
+- An asymmetric key pair (`PUB_AUTH_KEY`, `PRIV_AUTH_KEY`)
+
+  When using a password, the private key is generated by using the function `PBKDF_B` with the password
+
+  > A key property here is we only need the password to generate the required keys.
+  > Another approach would be to use the password as a symmetric key to encrypt a randomly generated private key that is sent alongside the public key.
+  > But we would lose that property.
+
+  For FIDO2, the private key already exists in the FIDO2 device so we can use it directly.
+
+  > [!NOTE]
+  > Should verify if it's possible to use fido2 to encrypt random blob of data.
+
+- A symmetric key (`SYM_KEY`) that is generated client-side at the final step of the account creation.
+  It's encrypted with `PUB_AUTH_KEY` and then uploaded to the service.
+
+> [!CAUTION]
+> The output of `PBKDF_A` and `PBKDF_B` should be different to avoid leaking the private key.
+
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Alice ->> S: Provide its email for the account creation
+  S ->> Alice: Send an email with a unique code need to continue the process
+  alt Choose between password or fido2
+    Alice ->> Alice: Choose a password
+    Alice ->> Alice: Derive a secret using the password and `PBKDF_A`
+    Alice ->> Alice: Create `PRIV_AUTH_KEY`, `PUB_AUTH_KEY` using the password and `PBKDF_B`
+  else
+    Alice ->> Alice: Use the public key of the FIDO2 device as `PUB_AUTH_KEY`, the private key of FIDO2 device as `PRIV_AUTH_KEY`
+  end
+
+  Alice ->> Alice: Create the symmetric key `SYM_KEY`
+  Alice ->> Alice: Encrypt the key `SYM_KEY` with `PUB_AUTH_KEY`
+  Alice ->> S: Send the require data to finish the account creation (`enc(SYM_KEY)`, `PUB_AUTH_KEY`, `AUTH_SECRET`, email)
+```
+
+At the end of the process, the server should save the following information in a new entry:
+
+- A UUID (`user_id`)
```

Comment:
Maybe make explicit that this user_id is not related to any organization's user_id.

Reply:
I've added a note to be more specific